avocado/utils/vmimage.py

```diff
@@ -235,7 +238,7 @@ def __init__(self, version="[0-9]+", build="[0-9]+.[0-9]+", arch=DEFAULT_ARCH):
         self.url_old_images = (
             "https://archives.fedoraproject.org/pub/archive/fedora/linux/releases/"
         )
-        self.image_pattern = "Fedora-Cloud-Base-(?P<version>{version})-(?P<build>{build}).(?P<arch>{arch}).qcow2$"
+        self.image_pattern = "Fedora-Cloud-Base-Generic-(?P<version>{version})-(?P<build>{build}).(?P<arch>{arch}).qcow2$"
```

Comment:
I believe this would make accessing Fedora 39 and earlier images impossible, because they don't have `Generic` in them. See  https://dl.fedoraproject.org/pub/fedora/linux/releases/39/Cloud/x86_64/images/